force-app/main/default/classes/RecurringDonationsService.cls

```diff
@@ -0,0 +1,157 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2022
+* @description Recurring Donations Service class.
+*/
+public with sharing class RecurringDonationsService {
+    
+    private final String MONTHLY = 'Monthly';
+    private final String WEEKLY = 'Weekly';
+    private final String YEARLY = 'Yearly';
+
+    @TestVisible
+    private UTIL_Permissions permissions {
+        get {
+            if(permissions == null) {
+                permissions = UTIL_Permissions.getInstance();
+            }
+            return permissions;
+        }
+        set;
+    }
+    
+    @TestVisible
+    private RecurringDonationsSelector selector {
+        get {
+            if(selector == null) {
+                selector = new RecurringDonationsSelector();
+            }
+            return selector;
+        }
+        set { 
+            this.selector = value;
+        }
+    }
+    
+    /** 
+    * @author Salesforce.org
+    * @date 2022
+    * @description method to check recurring donation permissions.
+    * @return List<RecurringDonationsController.RecurringDonation> returns processed Recurring Donations.
+    */
+    private Boolean hasAccessToRecurringDonation() {
+        return permissions.canRead(npe03__Recurring_Donation__c.SObjectType, new Set<SObjectField>{
+            npe03__Recurring_Donation__c.npe03__Amount__c,
+            npe03__Recurring_Donation__c.npe03__Installment_Period__c,
+            npe03__Recurring_Donation__c.InstallmentFrequency__c,
+            npe03__Recurring_Donation__c.Day_of_Month__c
+        });
+    }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2022
+    * @description method to return Recurring Donation list.
+    * @return List<RecurringDonationsController.RecurringDonation> returns processed Recurring Donations.
+    */
+    public List<RecurringDonationsController.RecurringDonation> getData() {
```

Comment:
This function is specifically used for the Community controller displayed. Therefore, the function should be moved to the Controller class and set to private.